src/Microsoft.Diagnostics.Monitoring.EventPipe/Configuration/MonitoringSourceConfiguration.cs

```diff
@@ -28,12 +28,55 @@ public abstract class MonitoringSourceConfiguration
         public const string EventPipeProviderName = "Microsoft-DotNETCore-EventPipe";
         public const string SystemDiagnosticsMetricsProviderName = "System.Diagnostics.Metrics";
 
+        private bool _requestRundown = true;
+        private long _rundownKeyword = EventPipeSession.DefaultRundownKeyword;
+
         public static IEnumerable<string> DefaultMetricProviders => new[] { SystemRuntimeEventSourceName, MicrosoftAspNetCoreHostingEventSourceName, GrpcAspNetCoreServer };
 
         public abstract IList<EventPipeProvider> GetProviders();
 
-        public virtual bool RequestRundown { get; set; } = true;
+        public virtual bool RequestRundown
```

Comment:
I would replace the RequestRundown property with the RundownKeyword property. Any config where RequestRundown = true would become RundownKeyword = DefaultRundownKeyword, otherwise RundownKeyword = 0.